game/src/Game/Autoload/CursorManager.gd

```diff
@@ -0,0 +1,173 @@
+
+	func generate_new_res(base_res_index:int, resolution:Vector2) -> void:
+		# resolution wasn't in among the default, need to generate it ourselves
+		CursorSingleton.generate_resolution(cursor_name,base_res_index,resolution)
+
+	#only bother with this if the cursor is animated
+	func _process_cursor(delta:float, shape:Input.CursorShape=Input.CURSOR_ARROW) -> void:
+		timeToFrame -= delta
+		if(timeToFrame <= 0):
+			currentFrame = (currentFrame + 1) % len(sequence)
+			timeToFrame = timings[self.sequence[currentFrame]] 
+			set_hardware_cursor(currentFrame, shape)
+
+	func set_hardware_cursor(frame:int=0, shape:Input.CursorShape=Input.CURSOR_ARROW) -> void:
+		var texture : ImageTexture = frames[sequence[frame]]
+		var hotspot : Vector2 = hotspots[sequence[frame]]
+		Input.set_custom_mouse_cursor(texture,shape,hotspot)
+
+
+#TODO: This is set on game start, but we probably want this to be a video setting
+var preferred_res : Vector2 = Vector2(32,32)
+
+
+var activeCursor : CompatCursor
+var activeShape : Input.CursorShape
+
+#Shape > Cursor dictionnaries
+var currentCursors : Dictionary = {
+	Input.CURSOR_ARROW:null,
+	Input.CURSOR_BUSY:null,
+	Input.CURSOR_IBEAM:null
+}
+var queuedCursors : Dictionary = {
+	Input.CURSOR_ARROW:null,
+	Input.CURSOR_BUSY:null,
+	Input.CURSOR_IBEAM:null
+}
+var loaded_cursors : Dictionary = {}
+
+func load_cursors() -> void:
+	CursorSingleton.load_cursors()
+	for cursor_name : StringName in CursorSingleton.cursor_names:
+		var cursor : CompatCursor = CompatCursor.new(cursor_name)
+		cursor.set_resolution(preferred_res)
+		loaded_cursors[cursor_name] = cursor
+
+var mouseOverWindow : bool = false
+
+#Handle queued cursor changes and cursor animations
+func _process(delta) -> void:
+	#only attempt to update the mouse when this wont crash anything
+	if mouseOverWindow:
+		var mouseShape : Input.CursorShape = Input.get_current_cursor_shape()
+
+		for shape in currentCursors.keys():
+			if currentCursors[shape] != queuedCursors[shape] and queuedCursors[shape] != null:
+				currentCursors[shape] = queuedCursors[shape]
+				currentCursors[shape].set_hardware_cursor(0,shape)
+			elif currentCursors[shape] != queuedCursors[shape]:
+				currentCursors[shape] = queuedCursors[shape]
+				Input.set_custom_mouse_cursor(null,shape)
+
+		#The mouse's cursor shape changed (something like we started hovering over text)
+		# reset the current cursor's frame, then switch the active cursor
+		if mouseShape != activeShape:
+			#Current mouse type changed, need to make sure that if the cursor of this new type
+			# is animated, we are providing its frames instead of the frames of the previous active cursor
+			activeShape = mouseShape
+			if mouseShape in currentCursors and currentCursors[mouseShape] != null:
+				activeCursor = currentCursors[mouseShape]
+				activeCursor.reset()
+				activeCursor.set_hardware_cursor(0,activeShape)
+			else:
+				activeCursor = null
+				Input.set_custom_mouse_cursor(null,mouseShape)
```

Comment:
```suggestion
			activeCursor = currentCursors.get(activeShape, null)
			if activeCursor != null:
				activeCursor.reset()
				activeCursor.set_hardware_cursor(0, activeShape)
			else:
				Input.set_custom_mouse_cursor(null, activeShape)
```
We can use `currentCursors.get(activeShape, null)` to get the active shape's cursor if it exists and fallback to `null` otherwise, simplifying the logic and reducing the amount of different code in each branch. Also since `activeShape = mouseShape`, it's cleaner to stick to using just one of them after that point